pupil_src/shared_modules/head_pose_tracker/function/utils.py

```diff
@@ -59,6 +59,35 @@ def convert_matrix_to_extrinsic(extrinsic_matrix):
     return merge_extrinsics(rotation, translation)
 
 
+def rod_to_euler(rotation_pose):
+    """
+    :param rotation_pose: Compact Rodrigues rotation vector, representing
+    the rotation axis with its length encoding the angle in radians to rotate
+    :return: x,y,z: Orientation in the Pitch, Roll and Yaw axes as Euler angles
+    according to 'right hand' convention
+    """
+    # convert Rodrigues rotation vector to matrix
+    rot = cv2.Rodrigues(rotation_pose)[0]
+
+    # rotate 180 degrees in y- and z-axes (corresponds to yaw and roll) to align
+    # with right hand rule (relative to the coordinate system of the origin marker
+    rot_mat = np.array([[1, 0, 0], [0, -1, 0], [0, 0, -1]])
+    rot = np.matmul(rot, rot_mat)
+
+    # convert to euler angles
+    sin_y = np.sqrt(rot[0, 0] * rot[0, 0] + rot[1, 0] * rot[1, 0])
+    if not sin_y < 1e-6:
+        x = np.arctan2(rot[2, 1], rot[2, 2])
+        y = -np.arctan2(-rot[2, 0], sin_y)
+        z = -np.arctan2(rot[1, 0], rot[0, 0])
+    else:
+        x = np.arctan2(-rot[1, 2], rot[1, 1])
+        y = -np.arctan2(-rot[2, 0], sin_y)
+        z = 0
+
+    return np.rad2deg([x, y, z])
+
+
 def get_camera_pose(camera_extrinsics):
```

Comment:
```suggestion
def get_camera_pose_and_orientation(camera_extrinsics):
```
I suggest reflecting the change in the function name (and its references)

Reply:
'Pose' refers to position and orientation. Therefore, I think adding 'orientation' would be superfluous/incorrect, as Euler is merely another representation of orientation.

Reply:
ok, in this case, I suggest renaming the `camera_pose` and `camera_orientation` variables to better reflect their content.

Reply:
I have left camera_pose unchanged, since this contains orientation and translation, but renamed camera_orientation to euler_orientation. I think these variable names are now representative